packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -998,6 +1003,13 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 ///     // Do something
 ///   }
 ///
+///   @override
+///   Widget build(BuildContext context) => new Container();
+///
+/// }
+///
+/// class RouteAwareWidget extends StatefulWidget {
+///   State<RouteAwareWidget> createState() => new RouteAwareWidgetState();
 /// }
 /// ```
 class RouteObserver<T extends Route<dynamic>> extends NavigatorObserver {
```

Comment:
Having said that, if we don't want this in the framework, we can also just put it in the Firebase Analytics Plugin (and then duplicate it for the Google Analytics Plugin). However, my discussion with @collinjackson indicated that it makes sense to have this logic in the framework to share it across the plugins.